server/src/main/java/com/flipkart/varadhi/services/ProjectService.java

```diff
@@ -1,24 +1,89 @@
 package com.flipkart.varadhi.services;
 
 import com.flipkart.varadhi.common.VaradhiCache;
-import com.flipkart.varadhi.entities.Project;
 import com.flipkart.varadhi.common.exceptions.InvalidOperationForResourceException;
 import com.flipkart.varadhi.common.exceptions.ResourceNotFoundException;
+import com.flipkart.varadhi.entities.Project;
 import com.flipkart.varadhi.spi.db.MetaStore;
 import io.micrometer.core.instrument.MeterRegistry;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
+/**
+ * Service for managing projects in Varadhi.
+ * <p>
+ * This service provides methods for creating, retrieving, updating, and deleting projects,
+ * as well as managing a cache of projects for efficient access.
+ */
+@Slf4j
 public class ProjectService {
```

Comment:
// retain the original ProjectService. CRUD from DB. No cache.
// create a diff class CacheProjectSupplier/Provider/Getter. get(name) -> Proejct. implement the entitiyEventListener

Reply:
btw. CacheProjectSupplier / CachedTopicSupplier / etc.. all will have the same class. The pattern is to preload the data and rely on events.

so maybe a single class to abstract this makes sense. Then you can just do ReadCache\<Project\>, ReadCache\<Topic\>, etc.